### PR TITLE
Update @hook attribute in docblock

### LINKED ISSUE #859: Ensure all hooks use the `@hook` attribute in the docblock
### Describe the enhancement

We use [WordPress HookDoc](https://github.com/matzeeable/wp-hookdoc) to build our [documentation site](https://10up.github.io/distributor/) but this relies on all hooks using the `@hook` attribute in the docblock for the hook. In looking at other things, I noticed at least one hook that wasn't using that, so that hook does not show in our documentation site. We should audit all custom hooks we are using and ensure they all have the `@hook` attribute.

### Steps to Reproduce

No real reproduction steps here but need to go through the entire codebase, finding any custom actions or filters and making sure they have `@hook` set up properly

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by akshitsethi**
### Description of the Change

Closes #859 

For this PR, I audited the `do_action` and `apply_filters` used in the code to ensure that the `@hook` attribute is present and while doing so, also updated the docblock content wherever necessary.

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by akshitsethi** [comment]
Linter issues are for `application-passwords` plugin.

File: includes/classes/InternalConnections/NetworkSiteConnection.php [hunk 1]
```diff
@@ -1102,12 +1102,11 @@ public function update_content_via_rest( $new_post_id ) {
 		 * Allow filtering of the HTTP request args before updating content
 		 * via a REST API call.
 		 *
```
Comment (dinhtungdu) [inline includes/classes/InternalConnections/NetworkSiteConnection.php:1104]: This is missing the `@hook` tag.

File: includes/classes/PullListTable.php [hunk 2]
```diff
@@ -63,7 +63,9 @@ public function get_columns() {
 		/**
 		 * Filters the columns displayed in the pull list table.
 		 *
-		 * @param array $columns An associative array of column headings.
+		 * @param {array} $columns An associative array of column headings.
+		 *
+		 * @return {array} An associative array of column headings.
 		 */
```
Comment (dinhtungdu) [inline includes/classes/PullListTable.php]: This is missing `@hook`.

File: includes/classes/PullListTable.php [hunk 3]
```diff
@@ -389,8 +393,10 @@ public function single_row( $item ) {
 		/**
 		 * Filters the class used on the table row on the pull list table.
 		 *
-		 * @param string $class The class name.
-		 * @param \WP_Post $item The current post object.
+		 * @param {string}  $class The class name.
+		 * @param {WP_Post} $item  The current post object.
+		 *
+		 * @return {string} The class name.
 		 */
```
Comment (dinhtungdu) [inline includes/classes/PullListTable.php:404]: This is missing `@hook`.

File: includes/utils.php [hunk 4]
```diff
@@ -1019,13 +1021,13 @@ function get_rest_url( $blog_id, $post_id ) {
 	restore_current_blog();
 
 	/**
-	 * Allow filtering of the REST API URL used for pulling post contewnt,
+	 * Allow filtering of the REST API URL used for pulling post content,
```
Comment (dinhtungdu) [inline includes/utils.php]:
```suggestion
	 * Allow filtering of the REST API URL used for pulling post content.
```

Comment (dinhtungdu) [reply includes/utils.php]: This is also missing the `@hook` tag.

**Review by dinhtungdu — COMMENTED** [review]
@akshitsethi Thanks so much for the PR! The aligned comments look awesome! 🎉  There are some missing `@hook` tags, please check my comments for more detail.

**Comment by akshitsethi** [comment]
Thank you @dinhtungdu for the suggestions and pointing out the `@hook` I missed adding.

Request you to re-review the PR.

**Review by dinhtungdu — APPROVED** [review]
Thanks for the update @akshitsethi! This is LGTM!

**Pull request merged**